File: app/src/main/java/com/github/mrvilkaman/presentationlayer/fragments/testfrags/Frag5ScreenFragment.java
package com.github.mrvilkaman.presentationlayer.fragments.testfrags;

import com.github.mrvilkaman.presentationlayer.activities.ActivityCoreComponent;
import com.github.mrvilkaman.presentationlayer.fragments.core.BaseFragment;

public class Frag5ScreenFragment extends FragBaseScreenFragment {


	public static Frag5ScreenFragment open() {
		return new Frag5ScreenFragment();
	}

	@Override
	protected int getNumber() {
		return 5;
	}

	@Override
	protected BaseFragment nextFragment() {
		return null;
	}

	@Override
	public void daggerInject() {
		ActivityCoreComponent component = getComponent(ActivityCoreComponent.class);
		DaggerFragScreenComponent.builder().activityCoreComponent(component).build().inject(this);
	}
}

File: kotlinapp/src/main/java/com/github/mrvilkaman/ui/screens/testfrags/Frag5ScreenFragment.kt
package com.github.mrvilkaman.ui.screens.testfrags

class Frag5ScreenFragment : FragBaseScreenFragment() {

    override val number: Int
        get() = 5

    override val icon: Int
        get() = android.R.drawable.ic_media_pause

    override fun nextScreenKey(): String? = null

    companion object {

        fun open(): Frag5ScreenFragment {
            return Frag5ScreenFragment()
        }
    }
}
